firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -2626,10 +2629,19 @@ class BrowserViewController: UIViewController,
     }
 
     func handleQRCode() {
+        cancelEditMode()
         openBlankNewTab(focusLocationField: false, isPrivate: false)
         navigationHandler?.showQRCode(delegate: self)
     }
 
+    // MARK: - Toolbar Refactor Deeplink Helper Method.
+    private func cancelEditMode() {
+        if isToolbarRefactorEnabled {
```

Comment:
Nit: I think a guard would be better here and would be easier to remove once we remove the old toolbar code.